omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -1784,62 +1729,75 @@ def _rotate_in_place(self, end_pose, angle_threshold=m.DEFAULT_ANGLE_THRESHOLD):
+        obj,
+        pose_on_obj=None,
+        sampling_attempts=m.MAX_ATTEMPTS_FOR_SAMPLING_POSE_NEAR_OBJECT,
+        skip_obstacle_update=False,
+        **kwargs,
+    ):
         """
         Returns a 2d pose for the robot within in the range of the object and where the robot is not in collision with anything
 
         Args:
-            obj (StatefulObject): Object to sample a 2d pose near
-            pose_on_obj (Iterable of arrays or None): The pose to sample near
+            obj (StatefulObject or list of StatefulObject): object(s) to sample a 2d pose near
+            pose_on_obj (Iterable or list of Iterable): (pos, quat) Pose(s) to sample near.
+                If provided, must match the length of obj list if obj is a list
 
         Returns:
             2-tuple:
                 - 3-array: (x,y,z) Position in the world frame
                 - 4-array: (x,y,z,w) Quaternion orientation in the world frame
         """
-        with PlanningContext(self.env, self.robot, self.robot_copy, "simplified") as context:
-            for _ in range(m.MAX_ATTEMPTS_FOR_SAMPLING_POSE_NEAR_OBJECT):
-                if pose_on_obj is None:
-                    pos_on_obj = self._sample_position_on_aabb_side(obj)
-                    pose_on_obj = [pos_on_obj, th.tensor([0, 0, 0, 1])]
-
-                distance_lo, distance_hi = 0.0, 5.0
-                distance = (th.rand(1) * (distance_hi - distance_lo) + distance_lo).item()
-                yaw_lo, yaw_hi = -math.pi, math.pi
-                yaw = th.rand(1) * (yaw_hi - yaw_lo) + yaw_lo
-                avg_arm_workspace_range = th.mean(self.robot.arm_workspace_range[self.arm])
-                pose_2d = th.cat(
-                    [
-                        pose_on_obj[0][0] + distance * th.cos(yaw),
-                        pose_on_obj[0][1] + distance * th.sin(yaw),
-                        yaw + math.pi - avg_arm_workspace_range,
-                    ]
-                )
-                # Check room
-                obj_rooms = (
-                    obj.in_rooms
-                    if obj.in_rooms
-                    else [self.env.scene._seg_map.get_room_instance_by_point(pose_on_obj[0][:2])]
-                )
-                if self.env.scene._seg_map.get_room_instance_by_point(pose_2d[:2]) not in obj_rooms:
-                    indented_print("Candidate position is in the wrong room.")
-                    continue
+        distance_lo, distance_hi = m.BASE_POSE_SAMPLING_LOWER_BOUND, m.BASE_POSE_SAMPLING_UPPER_BOUND
+        yaw_lo, yaw_hi = -math.pi, math.pi
+        avg_arm_workspace_range = th.mean(self.robot.arm_workspace_range[self.arm])
 
-                if not self._test_pose(pose_2d, context, pose_on_obj=pose_on_obj, **kwargs):
-                    continue
+        target_pose = (
+            (self._sample_position_on_aabb_side(obj), self._get_reset_eef_pose()[self.arm][1])
+            if pose_on_obj is None
+            else pose_on_obj
+        )
```

Comment:
Shouldn't you reuse the logic from `_grasp`?
e.g. calling `get_grasp_poses_for_object_sticky` stc?